### ISSUE TITLE
Span Support for Socket SendTo and UdpClient 

### ISSUE THREAD

**Issue by billknye** [dotnet/corefx#28131]
Checking out the new Span based socket apis in 2.1 preview 1, I noticed the api of UdpClient doesn't have span based overloads.  

Falling back to Socket, it appears there too the SendTo api is missing any overloads for Span and friends.  

Digging deeper into [SocketPal.Windows](https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs#L203-L238), there's no SendTo supporting span there but [SocketPal.Unix](https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs#L706-L750) does seem to have some Span support.  

The corresponding receive methods are also lacking Span support.  

To add actual non-allocating support, it would also appear that endpoint serialization in these methods would need to be tweaked but that seems to be somewhat the point of [SocketAddress](https://github.com/dotnet/corefx/blob/master/src/Common/src/System/Net/SocketAddress.cs) so maybe calling code would need to pass the SocketAddress instead of the IPEndPoint/EndPoint?

Was this an intentional gap in the api or just something that wasn't priority / low use case?

**Comment by stephentoub** [comment]
> just something that wasn't priority / low use case?

Exactly. We've added hundreds of span/memory-based methods based on an analysis of what's most common / impactful, but we fully expect more needs to be uncovered and more APIs will be added in the future. If this particular case is valuable to you, please feel free to submit an API proposal.

**Comment by karelz** [comment]
`Socket.SendTo` is covered in #35861

Triage: We should give `UdpCient` some Span overloads. Let's use this issue to track such investigation & work.

**Comment by scalablecory** [comment]
triage: look at other networking apis and add a mega-issue to spanify them.
